includes/session_details.js

```diff
@@ -0,0 +1,607 @@
+          pages_visited_details: {
+              description: "The pages visited within this session and associated metrics.",
+              columns: {
+                  anonymised_user_agent_and_ip: "One way hash of a combination of the user's IP address and user agent. Multiple users may share a single anonymised_user_agent_and_ip.",
+                  page: "The URL of the page visited",
+                  page_entry_time: "Timestamp indicating when the page was entered.",
+                  page_exit_time: "Timestamp indicating when the page was exited and the next page visit in the session began. This will be NULL is the user left the site after this page.",
+                  duration: "The difference in seconds between page_entry_time and page_exit_time. This will be NULL is the user left the site after this page.",
+                  next_step: "String indicating whether, at the end of this funnel, the user 'Left site immediately after this' or 'Visited subsequent pages'",
+                  exit_page_flag: "Indicates is the user left the site after this page. This is included as there are a small number of instances where a user has no subsequent page visits but 'next step' field shows 'Visited subsequent pages'"
+              }
+          }
+        } 
+  }).query(ctx => `
+  
+/* 
+
+  The session_details table is an alternative to the standard sessions table that is produced as part of the dfe-analytics dataform pipeline.
+
+  The session_details table uses Google Analytics definition of session:
+
+  "A session is a group of user interactions with your website that take place within a given time frame. For example a single session can contain multiple page views, events, social interactions, and ecommerce transactions."
+  In the context of Google Analytics, a given timeframe refers to any sequence of user interactions on a website that occur continuously without a break of more than 30 minutes of inactivity, until the user navigates away from the site or until a new user signs in with the same anonymised_user_agent_and_ip.
+  Importantly, by this definition a session is unique to a single user.
+
+  This table is necessary to accurately calculate the following metrics of interest:
+  1. Number of total sessions
+  2. Number of user sessions
+  3. Session duration
+  4. Page depth
+  5. Page duration
+  6. User Journeys
+
+*/
+
+  -- Define the base CTE with necessary fields and filters from the events dataset
+WITH
+  events AS (
+  SELECT
+    anonymised_user_agent_and_ip,
+    -- Unique identifier for user's ip/device combination
+    request_user_id,
+    -- User ID for the session (only available if the user signs in)
+    occurred_at,
+    -- Timestamp of the web request
+    request_path,
+    -- Path requested by the user
+    REGEXP_EXTRACT(request_referer, r'https?:\/\/([^\/]+)') AS request_referer_domain,
+    REGEXP_EXTRACT(request_referer, r'https?:\/\/[^\/]+(\/.*)') AS request_referer_path_and_query,
```

Comment:
And this logic too

Reply:
Added field to events table and updated the sessions detail table to use the new fields in the events table.